decision-records/2024-07-22--tempo-receivers-follow-up.md

```diff
@@ -0,0 +1,43 @@
+**Date:** 2024-07-22<br/>
+**Author:** Michael Dmitry (@michaeldmitry)  
+
+## Context and Problem Statement
+This is a follow-up on the ADR on [2024-07-05--tempo-receivers](2024-07-05--tempo-receivers.md#Decision) to decide which receivers would Tempo-k8s support. Tempo-k8s ought to support all receivers that upstream Tempo supports. After testing with different protocols from [opentelemetry receivers](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/) and referring to [Grafana Tempo source code](https://github.com/grafana/tempo/blob/main/modules/distributor/receiver/shim.go#L163C2-L169C3), the list of receivers supported by upstream Tempo comes down to 9 protocols: `zipkin`, `otlp_grpc`, `otlp_http`, `jaeger_grpc`, `opencensus`, `jaeger_thrift_http`, `kafka`, `jaeger_thrift_compact`, `jaeger_thrift_binary`.
+
+### Problem 1
+`kafka` protocol requires a running Kafka broker to run. If this protocol gets enabled in Tempo under `receivers` and we don't provide the address of a running broker, the Tempo workload will fail to start.
```

Comment:
can we link to a piece of documentation or something that backs this up?

Reply:
So, we'll not be able to find a documentation for this. This was found during experimentation from an error log describing whats above. The underlying reason is that Kafka receiver is using [Sarama](https://github.com/IBM/sarama) as a kafka consumer for the receiver that automatically dials the broker address once initialised, which then it doesn't find. Thus, fails.

Reply:
let's do that, otherwise tomorrow we'll forget how we know :)

Reply:
done